### PR TITLE
CORE-9083: add an endpoint to look up multiple subjects by ID.

### PR DESCRIPTION
**Pull request by slr71**

### CODE REVIEW

File: src/iplant_groups/clients/grouper.clj [hunk 1]
```diff
@@ -613,6 +613,30 @@
   (when (seq privilege-names)
     (add-folder-privileges username folder-name subject-ids privilege-names)))
 
+;; General subject functions.
+
+(defn- subject-id-lookup
+  [subject-id]
+  (remove-vals nil? {:subjectId subject-id}))
```
Comment (slr71) [inline src/iplant_groups/clients/grouper.clj:620]: This function was moved from below.

File: src/iplant_groups/routes/subjects.clj [hunk 2]
```diff
@@ -8,24 +8,32 @@
...
-        :query       [params SearchParams]
-        :return      SubjectList
-        :summary     "Subject Search"
-        :description "This endpoint allows callers to search for subjects by name."
-        (ok (subjects/subject-search params)))
+    :query       [params SearchParams]
+    :return      SubjectList
+    :summary     "Subject Search"
+    :description "This endpoint allows callers to search for subjects by name."
+    (ok (subjects/subject-search params)))
```
Comment (slr71) [inline src/iplant_groups/routes/subjects.clj:15]: This is just an indentation change.

File: src/iplant_groups/routes/subjects.clj [hunk 3]
```diff
@@ -8,24 +8,32 @@
...
-        :return      Subject
-        :summary     "Get Subject Information"
-        :description "This endpoint allows callers to get information about a single subject."
-        (ok (subjects/get-subject subject-id params)))
+    :path-params [subject-id :- SubjectIdPathParam]
+    :query       [params StandardUserQueryParams]
+    :return      Subject
+    :summary     "Get Subject Information"
+    :description "This endpoint allows callers to get information about a single subject."
+    (ok (subjects/get-subject subject-id params)))
```
Comment (slr71) [inline src/iplant_groups/routes/subjects.clj:32]: Another indentation change.

File: src/iplant_groups/routes/subjects.clj [hunk 4]
```diff
@@ -8,24 +8,32 @@
...
-        :return      GroupList
-        :summary     "List Groups for a Subject"
-        :description "This endpoint allows callers to list all groups that a subject belongs to."
-        (ok (subjects/groups-for-subject subject-id params))))
+    :path-params [subject-id :- SubjectIdPathParam]
+    :query       [params GroupsForSubjectParams]
+    :return      GroupList
+    :summary     "List Groups for a Subject"
+    :description "This endpoint allows callers to list all groups that a subject belongs to."
+    (ok (subjects/groups-for-subject subject-id params))))
```
Comment (slr71) [inline src/iplant_groups/routes/subjects.clj:40]: Yet another indentation change.

File: src/iplant_groups/service/format.clj [hunk 5]
```diff
@@ -85,26 +87,39 @@
...
+          :id                (:id subject)
+          :name              (:name subject)
+          :first_name        (get-attribute "givenName")
+          :last_name         (get-attribute "sn")
+          :email             (get-attribute "mail")
+          :institution       (get-attribute "o")
+          :description       (get-attribute "description")
+          :source_id         (:sourceId subject)}
+         (remove-vals nil?)
+         (remove-vals empty?))))
```
Comment (slr71) [inline src/iplant_groups/service/format.clj:107]: This function was extracted from `format-subject` so that it could be reused.

**Review by ianmcorvidae — APPROVED** [review]
Generally LGTM -- the one thought I have is for the sake of our future selves or anyone else using this later, it might be good to explicitly note that IDs which aren't found aren't in the return value either (presumably in the swagger documentation somewhere)

**Comment by slr71** [comment]
Thanks for the review!

**Pull request merged**